evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -1164,51 +1172,17 @@ def test_automl_time_series_classification_threshold(
     )
     automl.search()
     assert isinstance(automl.data_splitter, TimeSeriesSplit)
-    if objective == "Log Loss Binary":
-        mock_optimize_threshold.assert_not_called()
-        assert automl.best_pipeline.threshold is None
-        mock_split_data.assert_not_called()
-    elif optimize and objective == "F1":
+    if optimize:
         mock_optimize_threshold.assert_called()
         assert automl.best_pipeline.threshold == 0.62
         mock_split_data.assert_called()
-        assert str(mock_split_data.call_args[0][2]) == problem_type
-    elif not optimize and objective == "F1":
+    else:
         mock_optimize_threshold.assert_not_called()
-        assert automl.best_pipeline.threshold == 0.5
         mock_split_data.assert_not_called()
-
-
-@pytest.mark.parametrize("objective", ["F1", "Log Loss Binary", "AUC"])
-@patch("evalml.objectives.BinaryClassificationObjective.optimize_threshold")
-@patch(
-    "evalml.pipelines.BinaryClassificationPipeline._encode_targets",
-    side_effect=lambda y: y,
-)
-@patch("evalml.pipelines.BinaryClassificationPipeline.score")
-@patch("evalml.pipelines.BinaryClassificationPipeline.fit")
-@patch("evalml.pipelines.BinaryClassificationPipeline.predict_proba")
-def test_tuning_threshold_objective(
-    mock_predict,
-    mock_fit,
-    mock_score,
-    mock_encode_targets,
-    mock_optimize_threshold,
-    objective,
-    X_y_binary,
-):
-    mock_optimize_threshold.return_value = 0.6
-    X, y = X_y_binary
-    mock_score.return_value = {objective: 0.5}
-    automl = AutoMLSearch(
-        X_train=X, y_train=y, problem_type="binary", objective=objective
-    )
-    automl.search()
-
-    if objective != "F1":
-        assert automl.best_pipeline.threshold is None
-    else:
-        assert automl.best_pipeline.threshold == 0.6
+        if objective == "Log Loss Binary":
+            assert automl.best_pipeline.threshold is None
+        else:
+            assert automl.best_pipeline.threshold == 0.5
```

Comment:
Ah, so when "optimize" is true, that means automl will have run in this test with `optimize_thresholds` true, in which case the mock optimization value is expected, regardless of what the objective was. 👍 

If you wanna go above and beyond haha, I bet we could move most of this test to cover `engine_base.py` code directly instead of being an automl-level test. We wrote this test before the engine concept existed.

Reply:
@dsherry What do you mean here?

Reply:
Sorry lol. First part was me explaining to myself what this test does so that I understand it. Second part was a suggested improvement which you can certainly ignore if you'd like.

My point was that technically this test is checking the behavior of `EngineBase.train_and_score_pipelines` and doesn't have anything to do with `AutoMLSearch` itself, right? I guess its _also_ checking that the threshold values get saved and attached to `AutoMLSearch.best_pipeline`, but we could write a simpler test to check that using mocking.

Reply:
@dsherry I might just leave this test as is since it is particular to time series as well, and it seems to be a thorough enough test that shouldn't take much time. Seems like this test is just making sure that AutoMLSearch can threshold time series problems properly when needed.